File: 3143-longest-unequal-adjacent-groups-subsequence-i/3143-longest-unequal-adjacent-groups-subsequence-i.py
class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        ans, cur_group = [], -1
        for word, group in zip(words, groups):
            if group != cur_group:
                ans.append(word)
                cur_group = group
        return ans

File: 3143-longest-unequal-adjacent-groups-subsequence-i/3143-longest-unequal-adjacent-groups-subsequence-i.cpp
class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
        vector<string> ans;
        int n = groups.size();

        int prevGrp = groups[0];
        ans.push_back(words[0]);
        for(int i=1; i<n; i++){
            int curGrp = groups[i];
            if(curGrp!=prevGrp){
                ans.push_back(words[i]);
                prevGrp^=1;
            }
        }
        return ans;
    }
};